src/engine/source/builder/builders/opBuilderHelperMap.cpp

```diff
@@ -701,6 +705,60 @@ base::Expression opBuilderHelperAppendString(const std::any& definition)
         });
 }
 
+// field: +s_to_array/$field/[,| | ...]
+base::Expression opBuilderHelperAppendSplitString(const std::any& definition)
+{
+    auto [targetField, name, rawParameters] = helper::base::extractDefinition(definition);
+    auto parameters = helper::base::processParameters(rawParameters);
+    helper::base::checkParametersSize(parameters, 2);
+    helper::base::checkParameterType(parameters[0],
+                                     helper::base::Parameter::Type::REFERENCE);
+    helper::base::checkParameterType(parameters[1], helper::base::Parameter::Type::VALUE);
+    if (parameters[1].m_value.size() != 1)
+    {
+        throw std::runtime_error(fmt::format(
+            "[opBuilderHelperAppendSplit] separator can only be one character"));
+    }
+
+    name = helper::base::formatHelperFilterName(name, targetField, parameters);
+
+    // Tracing
+    const auto successTrace = fmt::format("[{}] -> Success", name);
+
+    const auto failureTrace1 =
+        fmt::format("[{}] -> Failure: parameter reference [{}] not found or not string",
+                    name,
+                    parameters[0].m_value);
+    const auto failureTrace2 = fmt::format("[{}] -> Failure", name);
+
+    // Return result
+    return base::Term<base::EngineOp>::create(
+        name,
+        [=, targetField = std::move(targetField)](
+            base::Event event) -> base::result::Result<base::Event>
+        {
+            auto resolvedReference = event->getString(parameters[0].m_value);
+            if (!resolvedReference.has_value())
+            {
+                return base::result::makeFailure(event, failureTrace1);
+            }
+
+            auto splitted =
+                utils::string::split(resolvedReference.value(), parameters[1].m_value[0]);
+            if (splitted.empty())
+            {
+                return base::result::makeFailure(event, failureTrace2);
+            }
```

Comment:
The array will have at least one element, if the character does not exist in the string, then the split function must return the array with only one element (the original string).